### PR TITLE
Delete membership for potential projects in dashboard [PEOPLE-96]

### PR DESCRIPTION
**Pull request by smnkrt**
JIRA: https://netguru.atlassian.net/browse/PEOPLE-96

In dashboard archive button should:
  -  set end date for selected membership when project is not potential
  - delete membership if project is potential

**Changes:**
 - add delete method to MembershipMailer with template
 - add project_potential field to membership/show.rable
 - add delete method to MembershipSource and MembershipStore
 - when archiving check if project associated with membership is potential

### CODE REVIEW

File: app/react/stores/MembershipStore.js [hunk 1]
```diff
@@ -41,6 +41,21 @@ class MembershipStore {
...
+  delete(params) {
+    const deleteMembership = (data) => {
+      const membershipsIds = this.memberships.map(membership => membership.id);
+      const membershipIndex = membershipsIds.indexOf(data.id);
+      this.memberships.splice(membershipIndex, 1)
+      Messenger().success("Membership deleted!");
+      this.emitChange();
+    };
+    const failedToDelete = () => {
+      console.log("FAILED TO DELETE");
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: Mixed double and single quotes.

File: app/react/stores/MembershipStore.js [hunk 2]
```diff
@@ -41,6 +41,21 @@ class MembershipStore {
...
 
+  delete(params) {
+    const deleteMembership = (data) => {
+      const membershipsIds = this.memberships.map(membership => membership.id);
+      const membershipIndex = membershipsIds.indexOf(data.id);
+      this.memberships.splice(membershipIndex, 1)
+      Messenger().success("Membership deleted!");
+      this.emitChange();
+    };
+    const failedToDelete = () => {
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: app/react/stores/MembershipStore.js [hunk 3]
```diff
@@ -41,6 +41,21 @@ class MembershipStore {
     return false;
   }
 
+  delete(params) {
+    const deleteMembership = (data) => {
+      const membershipsIds = this.memberships.map(membership => membership.id);
+      const membershipIndex = membershipsIds.indexOf(data.id);
+      this.memberships.splice(membershipIndex, 1)
+      Messenger().success("Membership deleted!");
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: Missing 'new' prefix when invoking a constructor.<br>Mixed double and single quotes.

File: app/react/stores/MembershipStore.js [hunk 4]
```diff
@@ -41,6 +41,21 @@ class MembershipStore {
     return false;
   }
 
+  delete(params) {
+    const deleteMembership = (data) => {
+      const membershipsIds = this.memberships.map(membership => membership.id);
+      const membershipIndex = membershipsIds.indexOf(data.id);
+      this.memberships.splice(membershipIndex, 1)
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: Missing semicolon.

File: app/react/stores/MembershipStore.js [hunk 5]
```diff
@@ -41,6 +41,21 @@ class MembershipStore {
     return false;
   }
 
+  delete(params) {
+    const deleteMembership = (data) => {
+      const membershipsIds = this.memberships.map(membership => membership.id);
+      const membershipIndex = membershipsIds.indexOf(data.id);
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).

File: app/react/stores/MembershipStore.js [hunk 6]
```diff
@@ -41,6 +41,21 @@ class MembershipStore {
     return false;
   }
 
+  delete(params) {
+    const deleteMembership = (data) => {
+      const membershipsIds = this.memberships.map(membership => membership.id);
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: app/react/stores/MembershipStore.js [hunk 7]
```diff
@@ -41,6 +41,21 @@ class MembershipStore {
     return false;
   }
 
+  delete(params) {
+    const deleteMembership = (data) => {
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: app/react/sources/MembershipSource.js [hunk 8]
```diff
@@ -30,4 +30,15 @@ export default class MembershipSource {
       }
     });
   }
+
+  static delete(params) {
+    return $.ajax({
+      url: Routes.membership_path(params.id),
```
Comment (netguru-hound) [inline app/react/sources/MembershipSource.js]: Identifier 'membership_path' is not in camel case.<br>'Routes' is not defined.

File: app/mailers/membership_mailer.rb [hunk 9]
```diff
@@ -19,4 +19,13 @@ def updated(data, current_user)
     subject = "Time span for #{@user.name} in #{@project.name} has been changed"
     mail(to: to, subject: subject, project: @project)
   end
+
+  def deleted(data, current_user)
+    @user = data[:user].decorate
+    @project = data[:project]
+    @current_user = current_user.decorate
+    to = mail_receivers(current_user, @user)
+    subject = "Membership for user #{@user.name} in potential project #{@project.name} has been deleted"
```
Comment (netguru-hound) [inline app/mailers/membership_mailer.rb]: Line is too long. [104/100]

File: app/react/stores/MembershipStore.js [hunk 10]
```diff
@@ -31,16 +31,31 @@ class MembershipStore {
...
+      const membershipsIds = this.memberships.map(membership => membership.id);
+      const membershipIndex = membershipsIds.indexOf(data.id);
+      this.memberships.splice(membershipIndex, 1);
+      Messenger().success('Membership deleted!');
+      this.emitChange();
+    };
+    const failedToDelete = () => {
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: 'const' is available in ES6 (use 'esversion: 6') or Mozilla JS extensions (use moz).<br>'arrow function syntax (=>)' is only available in ES6 (use 'esversion: 6').

File: app/react/stores/MembershipStore.js [hunk 11]
```diff
@@ -31,16 +31,31 @@ class MembershipStore {
...
     MembershipSource.update(params).done(updatedMembership).fail(failedToUpdate);
     return false;
   }
 
+  delete(params) {
+    const deleteMembership = (data) => {
+      const membershipsIds = this.memberships.map(membership => membership.id);
+      const membershipIndex = membershipsIds.indexOf(data.id);
+      this.memberships.splice(membershipIndex, 1);
+      Messenger().success('Membership deleted!');
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: Missing 'new' prefix when invoking a constructor.

File: app/react/stores/MembershipStore.js [hunk 12]
```diff
@@ -31,16 +31,31 @@ class MembershipStore {
       const membershipsIds = memberships.map(membership => membership.id);
       const membershipIndex = membershipsIds.indexOf(data.id);
       memberships[membershipIndex] = data;
-      Messenger().success("Membership updated!");
+      Messenger().success('Membership updated!');
```
Comment (netguru-hound) [inline app/react/stores/MembershipStore.js]: Missing 'new' prefix when invoking a constructor.

**Comment by konto-andrzeja** [comment]
Review done :+1:

**Pull request merged**